glib/src/translate.rs

```diff
@@ -501,69 +502,116 @@ impl<'a, P: Ptr, T: ?Sized + ToGlibPtr<'a, P>> ToGlibPtr<'a, P> for &'a T {
 }
 
 impl<'a> ToGlibPtr<'a, *const c_char> for str {
-    type Storage = CString;
+    type Storage = Cow<'static, [u8]>;
 
-    #[inline]
     fn to_glib_none(&'a self) -> Stash<'a, *const c_char, Self> {
-        let tmp =
-            CString::new(self).expect("str::ToGlibPtr<*const c_char>: unexpected '\0' character");
-        Stash(tmp.as_ptr(), tmp)
+        let bytes = if self.is_empty() {
+            Cow::Borrowed([0u8].as_slice())
+        } else {
+            if cfg!(debug_assertions) {
+                crate::GStr::check_interior_nuls(self).unwrap();
+            }
+            let mut bytes = self.as_bytes().to_owned();
+            bytes.reserve_exact(1);
```

Comment:
This potentially requires a re-allocation. Can we avoid that by pre-allocating the bytes with enough capacity?